internal/services/nginx/nginx_dataplane_apikey_resource.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nginx
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2024-11-01-preview/nginxapikey"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2024-11-01-preview/nginxdeployment"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type DataplaneAPIKeyModel struct {
+	Name              string `tfschema:"name"`
+	NginxDeploymentId string `tfschema:"nginx_deployment_id"`
+	SecretText        string `tfschema:"secret_text"`
+	Hint              string `tfschema:"hint"`
+	EndDateTime       string `tfschema:"end_date_time"`
+}
+
+type DataplaneAPIKeyResource struct{}
+
+var _ sdk.ResourceWithUpdate = (*DataplaneAPIKeyResource)(nil)
+
+func (m DataplaneAPIKeyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: nginxdeployment.ValidateNginxDeploymentID,
+		},
+
+		"secret_text": {
```

Comment:
What is the behavior if the secret_text changes? is that technically a new API Key or an update to the existing one?

Reply:
this should determine if you need to `ForceNew: true` on this argument.